python/lsst/pipe/base/configOverrides.py

```diff
@@ -125,35 +227,77 @@ def applyTo(self, config):
-                        value = ast.literal_eval(value)
-                    except Exception:
-                        # eval failed, wrap exception with more user-friendly
-                        # message
-                        raise pexExceptions.RuntimeError(f"Unable to parse `{value}' into a Python object")
-
-                # this can throw in case of type mismatch
-                setattr(obj, field[-1], value)
+                if isinstance(value, str):
+                    value = self._parser(value, configParser)
+                # checking for dicts and lists here is needed because {} and []
+                # are valid yaml syntax so they get converted before hitting
+                # this method, so we must parse the elements.
+                #
+                # The same override would remain a string if specified on the
+                # command line, and is handled above.
+                if isinstance(value, dict):
+                    new = {}
+                    for k, v in value.items():
+                        if isinstance(v, str):
+                            new[k] = self._parser(v, configParser)
+                        else:
+                            new[k] = v
+                    value = new
+                if isinstance(value, list):
+                    new = []
+                    for v in value:
+                        if isinstance(v, str):
+                            new.append(self._parser(v, configParser))
+                        else:
+                            new.append(v)
+                    value = new
+                # The field might be a string corresponding to a attribute
+                # hierarchy, attempt to split off the last field which
+                # will then be set.
+                parent, *child = field.rsplit(".", maxsplit=1)
+                if child:
+                    # This branch means there was a hierarchy, get the
+                    # field to set, and look up the sub config for which
+                    # it is to be set
+                    finalField = child[0]
+                    tmpConfig = attrgetter(parent)(config)
+                else:
+                    # There is no hierarchy, the config is the base config
+                    # and the field is exactly what was passed in
+                    finalField = parent
+                    tmpConfig = config
+                # set the specified config
+                setattr(tmpConfig, finalField, value)
+
             elif otype is OverrideTypes.Python:
-                exec(override, None, {"config": config})
+                # exec python string with the context of all vars known. This
+                # both lets people use a var they know about (maybe a bit
+                # dangerous, but not really more so than arbitrary python exec,
+                # and there are so many other places to worry about security
```

Comment:
Sounds like "let someone else worry about it" 😄

Reply:
Yeah kind of, that is how it has been explained to me. If we are using pex config to defend against code attacks, so many other layers of computer infrastructure have failed, and people would likely be able to do anything they want anyway.